File: src/Domino.Domain/Entities/GameRules.cs
namespace Domino.Domain.Entities;

public class GameRules
{
    public int MaxGrabsInRow { get; set; }
    public int MinLeftInMarket { get; set; }
    public string[] StarterTiles { get; set; } = [];
    public string[] HuntStarterTiles { get; set; } = [];
    public int PointsToStartHunt { get; set; }
    public bool WorkGoat { get; set; }
    public int TotalPointsToLoseWithGoat { get; set; }
    public Dictionary<string, int> LastTilePoints { get; set; } = [];
    public Dictionary<string, int> MorePointToEndWith { get; set; } = [];
}

File: src/Domino.UI/src/app/_models/gameRules.ts
export interface GameRules {
    maxGrabsInRow: number;
    minLeftInMarket: number;
    starterTiles: string[];
    huntStarterTiles: string[];
    pointsToStartHunt: number;
    workGoat: boolean;
    totalPointsToLoseWithGoat: number;
    lastTilePoints: Record<string, number>;
    morePointToEndWith: Record<string, number>;
}